SwiftCompilerSources/Sources/Optimizer/FunctionPasses/AllocVectorLowering.swift

```diff
@@ -259,7 +260,8 @@ private func createStackAllocatedVector(
   _ context: FunctionPassContext
 ) {
   let builder = Builder(before: allocVectorBuiltin, context)
-  let elementType = allocVectorBuiltin.substitutionMap.replacementTypes[0]!
+  let function = allocVectorBuiltin.parentFunction
+  let elementType = allocVectorBuiltin.substitutionMap.replacementTypes[0].loweredType(in: function)
```

Comment:
Maybe for these cases, add a `var replacementType` computed property which asserts that the length is exactly one and then returns the sole element?